scripts/cleanup_old_vms.py

```diff
@@ -70,6 +70,32 @@ def process_provider_vms(provider_key, matchers, delta, vms_to_delete):
         logger.exception(ex)
 
 
+def delete_provider_vms(provider_key, vm_names):
+    with lock:
+        print 'Deleting VMs from {} ...'.format(provider_key)
+
+    try:
+        with lock:
+            provider = get_mgmt(provider_key)
+    except Exception as e:
+        with lock:
+            print "Could not retrieve the provider {}'s mgmt system ({}: {})".format(
+                provider_key, type(e).__name__, str(e))
+            logger.exception(e)
+
+    for vm_name in vm_names:
+        with lock:
+            print "Deleting {} from {}".format(vm_name, provider_key)
+        try:
+            provider.delete_vm(vm_name)
```

Comment:
Eell, the provider object is created inside the new thread, so it should be a separate instance. (line 79)